lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -88,21 +87,19 @@ def _calculate_entrance_exam_score(user, course_descriptor, exam_modules):
     """
     Calculates the score (percent) of the entrance exam using the provided modules
     """
-    # All of the exam module ids
-    exam_module_ids = [exam_module.location for exam_module in exam_modules]
-
-    # All of the corresponding student module records
-    student_modules = StudentModule.objects.filter(
-        student=user,
-        course_id=course_descriptor.id,
-        module_state_key__in=exam_module_ids,
-    )
     student_module_dict = {}
-    for student_module in student_modules:
-        student_module_dict[unicode(student_module.module_state_key)] = {
-            'grade': student_module.grade,
-            'max_grade': student_module.max_grade
-        }
+    from courseware.grades import field_data_cache_for_grading
+    field_data_cache = field_data_cache_for_grading(course_descriptor, user)
+    scores_client = ScoresClient.from_field_data_cache(field_data_cache)
```

Comment:
I think you can just use `ScoresClient(course_descriptor.id, user.id)` and then `fetch_scores(exam_module_ids)`.